test/tests/bootstrap/test_api20_os_bootstrap_parallel_local.py

```diff
@@ -0,0 +1,297 @@
+on-http/static/http/repo/centos/6.5
+on-http/static/http/repo/centos/7.0
+on-http/static/http/repo/rhel/7.0
+on-http/static/http/repo/suse/42.1
+on-http/static/http/repo/esxi/5.5
+on-http/static/http/repo/esxi/6.0
+on-http/static/http/repo/winpe
+on-http/static/http/repo/coreos/899.17.0
+
+'''
+
+import fit_path  # NOQA: unused import
+from nose.plugins.attrib import attr
+import fit_common
+import flogging
+import time
+log = flogging.get_loggers()
+
+# This gets the list of nodes
+NODECATALOG = fit_common.node_select()
+
+# dict containing bootstrap workflow IDs and states
+NODE_STATUS = {}
+
+# global timer
+START_TIME = time.time()
+
+# collect repo information from config files
+OSLIST = fit_common.fitcfg()["install-config"]["os-install"]
+
+# download RackHD config from host
+rackhdconfig = fit_common.rackhdapi('/api/2.0/config')['json']
+rackhdhost = "http://" + str(rackhdconfig['apiServerAddress']) + ":" + str(rackhdconfig['apiServerPort'])
+
+
+# this routine polls a workflow task ID for completion
+def wait_for_workflow_complete(taskid):
+    result = None
+    while time.time() - START_TIME < 1800 or result is None:  # limit test to 30 minutes
+        result = fit_common.rackhdapi("/api/2.0/workflows/" + taskid)
+        if result['status'] != 200:
+            log.error(" HTTP error: " + result['text'])
+            return False
+        if result['json']['status'] == 'running' or result['json']['status'] == 'pending':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            fit_common.time.sleep(30)
+        elif result['json']['status'] == 'succeeded':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            return True
+        else:
+            log.error(" Workflow failed: " + result['text'])
+            return False
+    log.error(" Workflow Timeout: " + result['text'])
+    return False
+
+
+# helper routine to return the task ID associated with the running bootstrap workflow
+def node_taskid(workflow, version, kvm):
+    for entry in NODE_STATUS:
+        if NODE_STATUS[entry]['workflow'] == workflow \
+                and str(version) in NODE_STATUS[entry]['version'] \
+                and NODE_STATUS[entry]['kvm'] == kvm:
+            return NODE_STATUS[entry]['id']
+    return ""
+
+
+# Match up tests to node IDs to feed skip decorators
+index = 0  # node index
+for item in OSLIST:
+    if index < len(NODECATALOG):
+        NODE_STATUS[NODECATALOG[index]] = \
+            {"workflow": item['workflow'], "version": item['version'], "kvm": item['kvm'], "id": "Pending"}
+    index += 1
+
+# ------------------------ Tests -------------------------------------
+
+
+@attr(all=True)
+class api20_bootstrap_base(fit_common.unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # run all OS install workflows first
+        nodeindex = 0
```

Comment:
Add a check in here to make sure NODECATALOG is not [] and if so, log an error for the user and be done.
if NODECATALOG:  for item in OSLIST: ...  else log.error("no nodes exists in testbed") ....along that line.